jscomp/others/bs_Set.mli

```diff
@@ -0,0 +1,172 @@
+
+
+
+type ('elt, 'id) t0
+
+type ('elt, 'id) t = {
+  cmp : ('elt,'id) Bs_Cmp.t ; 
+  data : ('elt,'id) t0
+}
+(** The type of sets. *)
+
+val empty0: ('elt, 'id) t0
+val empty : ('elt, 'id) Bs_Cmp.t -> ('elt, 'id) t
+(** The empty set. *)
+
+val isEmpty0: ('elt, 'id) t0 -> bool
+val isEmpty : ('elt, 'id) t -> bool
+(** Test whether a set is empty or not. *)
+
+val mem0: 
+  cmp: ('elt,'id) Bs_Cmp.cmp ->
+  'elt -> ('elt, 'id) t0 -> bool
+val mem:  
+  'elt -> ('elt, 'id) t -> bool
+(** [mem x s] tests whether [x] belongs to the set [s]. *)
+
+val add0: 
+  cmp: ('elt,'id) Bs_Cmp.cmp ->
+  'elt -> ('elt, 'id) t0 -> ('elt, 'id) t0
+val add:   
+  'elt -> ('elt, 'id) t -> ('elt, 'id) t
+(** [add x s] returns a set containing all elements of [s],
+    plus [x]. If [x] was already in [s], [s] is returned unchanged. *)
+
+val singleton0: 'elt -> ('elt, 'id) t0
+val singleton : 
+  ('elt,'id) Bs_Cmp.t -> 
+  'elt -> ('elt, 'id) t
+(** [singleton x] returns the one-element set containing only [x]. *)
+
+val remove0: 
+  cmp: ('elt,'id) Bs_Cmp.cmp ->
+  'elt -> ('elt, 'id) t0 -> ('elt, 'id) t0
+val remove:  
+  'elt -> ('elt, 'id) t -> ('elt, 'id) t
+(** [remove x s] returns a set containing all elements of [s],
+    except [x]. If [x] was not in [s], [s] is returned unchanged. *)
+
+val union0: 
+  cmp: ('elt,'id) Bs_Cmp.cmp ->
+  ('elt, 'id) t0 -> ('elt, 'id) t0 -> ('elt, 'id) t0
+val union:  
+  ('elt, 'id) t -> ('elt, 'id) t -> ('elt, 'id) t
+(** Set union. *)
+
+val inter0: 
+  cmp: ('elt,'id) Bs_Cmp.cmp ->
+  ('elt, 'id) t0 -> ('elt, 'id) t0 -> ('elt, 'id) t0
+val inter: 
+  ('elt, 'id) t -> ('elt, 'id) t -> ('elt, 'id) t
+(** Set intersection. *)
+
+val diff0: 
+  cmp: ('elt,'id) Bs_Cmp.cmp ->
+  ('elt, 'id) t0 -> ('elt, 'id) t0 -> ('elt, 'id) t0
+val diff:   
+  ('elt, 'id) t -> ('elt, 'id) t -> ('elt, 'id) t
+(** Set difference. *)
+
+val cmp0: 
+  cmp: ('elt,'id) Bs_Cmp.cmp ->
+  ('elt, 'id) t0 -> ('elt, 'id) t0 -> int
+val cmp:  
```

Comment:
The corresponding operations for maps are called compare/equal.

Reply:
it is intentional to avoid conflict with compare, the thing is such unsafe `compare` is in scope by default..

Reply:
For consistency Bs_Map should also use the names cmp/eq. Currently it uses compare/equal.

Reply:
Yes, just updated